app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -220,28 +222,31 @@ function ($attribute, $value, $fail) {
 
             return response()->json(['error' => 'Failed to convert KML to GeoJSON', 'message' => $process->getErrorOutput()], 500);
         }
-
+        $geojsonContent = file_get_contents($geojsonPath);
+        $redis_key = 'kml_file_' . uniqid();
+        Redis::set($redis_key, $geojsonContent, 'EX', 7200);
+    
         $polygonLoadedList = isset($body['polygon_loaded']) && filter_var($body['polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
         $submitPolygonsLoaded = isset($body['submit_polygon_loaded']) && filter_var($body['submit_polygon_loaded'], FILTER_VALIDATE_BOOLEAN);
 
-        if (! $polygonLoadedList && ! $submitPolygonsLoaded) {
-            $job = new InsertGeojsonToDBJob($geojsonFilename, $site_id, 'site', $body['primary_uuid'] ?? null);
+        $delayedJob = DelayedJob::create(['status' => InsertGeojsonToDBJob::STATUS_PENDING]);
+
+        if (! $polygonLoadedList && !$submitPolygonsLoaded) {
+          $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
         }
 
         if ($polygonLoadedList) {
-            $geojsonContent = file_get_contents($geojsonPath);
-            $polygonLoaded = $this->GetAllPolygonsLoaded($geojsonContent, $site_id);
-
-            return response()->json($polygonLoaded->original, 200);
+          $polygonLoaded = $this->GetAllPolygonsLoaded($geojsonContent, $site_id);
+          return response()->json($polygonLoaded->original, 200);
         }
 
         if ($submitPolygonsLoaded) {
-            $job = new InsertGeojsonToDBJob($filename, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
+          $job = new InsertGeojsonToDBJob($redis_key, $delayedJob->id, $site_id, 'site', $body['primary_uuid'] ?? null, $body['submit_polygon_loaded']);
```

Comment:
It took me a second to decipher these three if blocks because at first glance it looks like a job can be created and then not used in the return response. I'd restructure a bit to make it more readable. Also, it looks like the job created on 235 and here are the same, and that a DelayedJob can end up getting created (and a Redis payload set) that doesn't get used.

```
if ($polygonLoadedList) {
  // ... return loaded response
} 

// insert payload to Redis, create DelayedJob and InsertGeojsonToDBJob
```

Reply:
restructure the code without creating the job.